sdks/python/container/Dockerfile

```diff
@@ -82,6 +78,32 @@ RUN  \
     # Remove pip cache.
     rm -rf /root/.cache/pip
 
+FROM gcr.io/distroless/cc-debian12 as beam
+LABEL Author "Apache Beam <dev@beam.apache.org>"
+ARG TARGETOS
+ARG TARGETARCH
+
+# copy commands & libs to distroless
+COPY --from=python-base /bin /bin
```

Comment:
for my information:

1. how did we choose what content to copy from base?
2. do distroless images have a package manager (`apt`) ?  Will users be able to install additional software into these images if they want?

Reply:
/bin contains some system commands like `ls`. We do not need them in theory.

no apt on distroless. If you are talking about the customer container, it will be better for users to use other images as base or use the way in this PR to copy the packages over.

Reply:
I think we have to announce this in breaking changes in CHANGES.MD and in make adjustments to custom container documentation in Beam / Dataflow docs. I do see that this could complicate the UX for some custom container users.

Reply:
Sounds good. I will update CHANGES.md after we all agree with using distroless.

Reply:
Not a blocker here, just a general comment: I'd rather pick specific binaries rather than copying every `/bin` binary. I understand it probably means more breakages, but it's also a smaller vulnerability surface.

Reply:
I think lack of `apt` is the main inconvenience for me. I think this is a real friction point, while the vulnerabilities reported by docker image checkers are more often than not, are not applicable to execution of Beam pipelines.

Reply:
Before custom containers we also mentioned running custom commands in: 

https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#nonpython

Not sure how much usage this has though.